### PR TITLE
[Merged by Bors] - Add documentation comments to `bevy_window`

### LINKED ISSUE #3492: Tracking issue: deny missing docs, one crate at a time
Docs are great! However, it's easy to let them atrophy.

As Bevy grows and stabilizes, we want to be able to turn on `#![warn(missing_docs)]` (which will cause CI to fail if violated). This is helpful because it helpful for new users and contributors, and ensures that our code base's documentation state doesn't regress.

However, not everything is stable or well documented enough to do so already.
We should turn this on one crate at a time, once it hits 100% coverage. If there's a crate

Let's check the current doc coverage.

1. Set the following environment variable (on Windows, I used the [`$Env:` syntax](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.2)): `RUSTDOCFLAGS="-Z unstable-options --show-coverage"`
2. Run `cargo +nightly doc --workspace --all-features --no-deps` 

Run on 2023-03-14:
**Crate | Docs Coverage** 

- [x] `bevy_a11y` | 100.0%
- [x] `bevy_animation` | 100.0%
- [ ] `bevy_utils` | 53.3%
- [x] `bevy_tasks` | 100.0% (#3509)
- [ ] `bevy_derive` | 33%
- [ ] `bevy_macro_utils` | 28.6%
- [x] `bevy_math` | 100.0% (#3503 | #4591)
- [x] `bevy_app` | 100.0% (#3539)
- [ ] `bevy_ecs_macros` | 40.0%
- [ ] `bevy_dynamic_plugin` | 42.9%
- [ ] `bevy_window` | 97.8% (#4333)
- [x] `bevy_crevice` | **100%** (Not our problem)
- [x] `bevy_log` | 100.0%
- [x] `bevy_transform` | 100.0%
- [x] `bevy_core` | 100.0%
- [ ] `bevy_reflect`| 61.7%
- [ ] `bevy_input` | 90.6% (Claimed by @KDecay)
- [ ] `bevy_gilrs` | 0%
- [ ] `bevy_diagnostic` | 50%
- [ ] `bevy_winit` | 56%
- [x] `bevy_audio` | 100.0%
- [ ] `bevy_scene` | 22.1%
- [x] `bevy_asset` | 100.0% (https://github.com/bevyengine/bevy/pull/3536)
- [ ] `bevy_core_pipeline` | 3.4%
- [ ] `bevy_gltf` |16.7%
- [ ] `bevy_sprite` | 30.8%
- [ ] `bevy_ecs` | 79.5%
- [x] `bevy_dylib` | **100%** (warning not yet enabled) (#3515)
- [ ] `bevy_text` | 35.8%
- [ ] `errors` | 66.7%
- [x] `bevy_internal` | 100.0% (#3514)
- [ ] `bevy_pbr` | 37.4%
- [ ] `bevy_ui` | 74.8%
- [ ] `bevy_render` | 48.2%
- [ ] `bevy_core_pipeline` | 22.3%
- [ ] `bevy_derive` | 57.1%
- [ ] `bevy_encase_derive` | 0%
- [x] `bevy_hierarchy` | 100%
- [x] `bevy_log` | 100%
- [ ] `bevy_mikktspace` | 90%
- [x] `bevy_ptr` | 100%
- [ ] `bevy_reflect_derive`| 88.9%
- [ ] `bevy_render_macros` | 25%
- [ ] `bevy_time` | 78.8%
- [ ] `bevy_utils_macros` | 0%

### PR DESCRIPTION
**Pull request by x-52**
# Objective
- Add documentation comments and `#![warn(missing_docs)]` to `bevy_window`.
- Part of #3492

### CODE REVIEW

File: crates/bevy_window/src/cursor.rs [hunk 1]
```diff
@@ -1,39 +1,79 @@
 /// The icon to display for a window's cursor
+/// Examples of all of these cursors can be found [here](https://www.w3schools.com/cssref/playit.asp?filename=playcss_cursor)
+/// This `enum` is simply a copy of a similar `enum` found in `winit`.
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/cursor.rs]: We should be able to link the winit crate here :)

File: crates/bevy_window/src/cursor.rs [hunk 2]
```diff
@@ -1,39 +1,79 @@
...
+/// `winit`, in turn, simply used cursor types avilable in the browser.
 #[derive(Debug, Hash, PartialEq, Eq, Clone, Copy)]
 pub enum CursorIcon {
+    /// The platform-dependent default cursor.
     Default,
+    /// A simple crosshair.   
     Crosshair,
+    /// A hand (often used to indicate links in web browsers).    
     Hand,
+    /// Self explanatory.
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/cursor.rs:13]: It's obvious, but not all readers have the same background.  What sort of arrow is it, and when is it used,?

File: crates/bevy_window/src/window.rs [hunk 3]
```diff
@@ -169,53 +171,45 @@ pub struct Window {
     pub canvas: Option<String>,
     command_queue: Vec<WindowCommand>,
 }
-
+/// A command to be sent to a window.
 #[derive(Debug)]
 pub enum WindowCommand {
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:176]: IMO we should have an example in a doc test here. Use no_run; this is just to verify that it compiles and demonstrate usage.

File: crates/bevy_window/src/window.rs [hunk 4]
```diff
@@ -557,49 +556,63 @@ impl Window {
...
     }
-
+    /// Get the [`RawWindowHandleWrapper`] corresponding to this window
     pub fn raw_window_handle(&self) -> RawWindowHandleWrapper {
         self.raw_window_handle.clone()
     }
 }
-
+/// A [`Window`]'s properties
 #[derive(Debug, Clone)]
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:589]: How do users access this in a typical Bevy app?

File: crates/bevy_window/src/lib.rs [hunk 5]
```diff
@@ -23,6 +24,7 @@ pub mod prelude {
 use bevy_app::prelude::*;
 use bevy_ecs::event::Events;
 
+/// A plugin that defines an interface for windowing support in Bevy.
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/lib.rs:27]: Probably worth linking the trait.

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
Nice work! Some light feedback, but this is well done.

File: crates/bevy_window/src/lib.rs (hunk 5 above)
Comment (alice-i-cecile) [reply crates/bevy_window/src/lib.rs:27]:
```suggestion
/// A [`Plugin`] that defines an interface for windowing support in Bevy.
```

File: crates/bevy_window/src/raw_window_handle.rs [hunk 6]
```diff
@@ -26,7 +26,7 @@ impl RawWindowHandleWrapper {
 // https://github.com/rust-windowing/raw-window-handle/issues/59
 unsafe impl Send for RawWindowHandleWrapper {}
 unsafe impl Sync for RawWindowHandleWrapper {}
-
+#[allow(missing_docs)] // Almost nobody uses this struct.
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/raw_window_handle.rs:29]:
```suggestion
#[doc(hidden)] // Add docs later; this is an obscure implementation detail
```

Comment (alice-i-cecile) [reply crates/bevy_window/src/raw_window_handle.rs:29]: Having poked at this more, I think this is the better approach for now.

File: crates/bevy_window/src/window.rs (hunk 3 above)
Comment (x-52) [reply crates/bevy_window/src/window.rs:177]: Bevy apps don't interact with `WindowCommand` directly. Instead, they use the methods on the `Window` `struct`. This `enum` is for crates that actually implement Bevy's windowing functionality (like `bevy_winit`). `bevy_window` is, as mentioned in an earlier commit, only a common interface for windowing functionality, so I don't think we need a doc test here.

Comment (x-52) [reply crates/bevy_window/src/window.rs:177]: I'm going to clarify this in a future commit.

File: crates/bevy_window/src/window.rs [hunk 7]
```diff
@@ -129,6 +131,7 @@ impl WindowResizeConstraints {
 }
 
 /// An operating system window that can present content and receive user input.
+/// To create a window, use a [`bevy_ecs::event::EventWriter`]`<`[`crate::CreateWindow`]`>`
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:134]:
```suggestion
/// To create a window, use a [`EventWriter<CreateWindow>`]`(`crate::CreateWindow).
```
I think this is sufficiently clear to anyone who's managed to stumble across these docs, and should do less strange things to the formatting.

File: crates/bevy_window/src/window.rs [hunk 8]
```diff
@@ -557,49 +576,76 @@ impl Window {
...
     }
-
+    /// Get the [`RawWindowHandleWrapper`] corresponding to this window
     pub fn raw_window_handle(&self) -> RawWindowHandleWrapper {
         self.raw_window_handle.clone()
     }
 }
-
+/// A [`Window`]'s properties
+/// # Creating a `WindowDescriptor`
```
Comment (alice-i-cecile) [inline crates/bevy_window/src/window.rs:609]:
```suggestion
///
/// # Creating a `WindowDescriptor`
///
```

**Review by alice-i-cecile — COMMENTED** [review]
Some *very* minor formatting nits, then this LGTM!

**Review by alice-i-cecile — APPROVED** [review]

**Review by nicopap — APPROVED** [review]
There are formatting issues. The first sentence of the doc string should be separated by an empty line from the rest, and all sentences should end with a point. Please, there is more to fix than what I added in suggestions (I gave up after the 15th iteration). Make sure to check the question on `WindowId::new` too. Those are probably blockers for the engine team, but not for me.

But otherwise I don't have any objections. Bonus points for doc example :+1:. (This is an approval)

**Review by Weibye — APPROVED** [review]
Approving this in its current state, awesome work! I'd prefer the nits were addressed, but not blocking. I would like this to get in before #4925

**Comment by alice-i-cecile** [comment]
@x-52 do you have time to tackle these last nits? If not we can remake this PR quick and credit you.

**Comment by x-52** [comment]
> @x-52 do you have time to tackle these last nits? If not we can remake this PR quick and credit you.

I'll do it right now.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Ah there's a tiny failure @x-52

**Pull request closed without merging**